### ISSUE TITLE
Best practice for provisioning external login account

### ISSUE THREAD

**Issue by levince10** [IdentityServer/IdentityServer4#546]
Hi,

I have a web application that can handle two main roles, let's say Student and Teacher for the example. I have also an IdSrv4 where they can login as a local account or with an external provider (Facebook).
For registration, both of roles can choose to create an account with a custom Registration page inside the Web app, or redirect to IdSrv to use Facebook to create the account.

If they choose Facebook for the first time, the ExternalLoginCallback will detect that they don't have an account yet in the system and try to create a user. (Or display a form to complete the registration first)
That would be fine IF i was not dealing with Roles.

Problem is: at this step of the game, I have no idea to know if the user is a Student or a Teacher. The Student registration form and the Teacher registration form are different and I need to display the right form to them.
I know we could do that by using different scopes, but that could work if both roles had their own web app.
I also thought about passing the role from the web app to IdSrv but I haven't found a way to do that and I am not sure I want to do that.

Here I feel that embedding some registration UI in IdSrv is not the best practice. It looks like I need to redirect the user as "Partially logged" (as IdSrv3), and then display the right form within the web app, since the web app knows for which role the user is trying to register.

What are your thoughts about this? Can you confirm or do you know some other options?

Thank you!

**Comment by leastprivilege** [comment]
sorry we are busy right now doing finishing touches on the RTM - still an issue?

**Comment by brockallen** [comment]
This is really more of a general question and not so much an issue or bug with IdentityServer. Perhaps stack over flow is a better place for such questions.

**Comment by levince10** [comment]
I decided to remove any "role business" in IdSrv and to leave the registration UI the responsability of the client. When registering with an external account, I just create an entry in the AspNetUsers table without any other role/business logic. When redirecting back to the client, the app forces the user to finish its registration by entering the necessary business information.
If the user tries to login to another app (AppB) with the same IdSrv, AppB will also force the user to enter its necessary business information so he can use AppB.
I found this solution much more scalable and maintainable. (No need to update constantly IdSrv when adding/modifying business logic for an app)

In the end I realize this might be the direction you wanted to lead us with IdSrv4 and if it's the case, then my bad for the confusion.
